### PR TITLE
fix Windows build

### PR DESCRIPTION
**Pull request by ssiloti**

### CODE REVIEW

File: src/file.cpp [hunk 1]
```diff
@@ -250,10 +276,6 @@ namespace libtorrent {
 #endif
 	}
 
-#ifndef INVALID_HANDLE_VALUE
-#define INVALID_HANDLE_VALUE (-1)
-#endif
```
Comment (arvidn) [inline src/file.cpp]: I think this might be used for non-windows builds.

**Comment by arvidn** [comment]
it also looks like the terms in between ``|`` needs to be parenthesized: https://github.com/arvidn/libtorrent/pull/2334/files#diff-216c6cad716c26d79607e6e3c7effa0fR96

**Comment by arvidn** [comment]
lgtm

**Comment by arvidn** [comment]
the failure looks like it might be related to the ``lock_files`` flag, preventing concurrent access.

**Comment by ssiloti** [comment]
At least one of the problems is that because we release the mutex in `file_view file_view_pool::open_file` while opening the file it, is possible for multiple threads to attempt to open the same file. Obviously only the first will succeed.

**Comment by ssiloti** [comment]
Even if we serialize opening files somehow, file locking still conflicts with the path which swaps a read only mapping for a read/write one. I don't see any way around this without making the swap non-atomic. Maybe file locking isn't worth the trouble?

**Comment by arvidn** [comment]
yeah, maybe not

**Comment by arvidn** [comment]
so, I guess file locking is just broken in master right now then.

**Comment by ssiloti** [comment]
On master `file_pool::open_file` holds its lock in the regular open file path so that's not a problem, but the read-only -> read/write upgrade path is broken if `lock_files` is enabled.

**Comment by ssiloti** [comment]
Ahhh, on master there is a fallback which re-opens the file without locking if the initial open fails. That would explain what we're not seeing failures.

**Comment by arvidn** [comment]
in order to land this patch, how about we just disable ``lock_files`` by default?

**Comment by arvidn** [comment]
looks like a mingw file handle mixup. https://ci.appveyor.com/project/arvidn/libtorrent/build/7241/job/f26a2yr90fb3p72o#L130
(I'm sorry I made a mess)

**Pull request merged**